File: tweeter-api/Models/Tweet.cs
using System.ComponentModel.DataAnnotations;

namespace tweeter_api.Models;

public class Tweet 
{
    public int TweetId { get; set; }
    [Required]
    public string? Title { get; set; }
    [Required]

    public string? TweetBody { get; set; }
    [Required]

    public string? TweeterId { get; set; }
}

File: frontend/tweeter-client/src/app/models/tweet.ts
export class Tweet {
  tweetId?: number;
  title?: string;
  createdAt?: Date;
  editedAt?: Date;
  tweetBody?: string;
  tweeterId?: number;

  constructor(
    id?: number,
    title?: string,
    tweetBody?: string,
    createdAt?: Date,
    editedAt?: Date,
    tweeterId?: number
  ) {
    this.tweetId = id;
    this.title = title;
    this.tweetBody = tweetBody;
    this.createdAt = createdAt;
    this.editedAt = editedAt;
    this.tweeterId = tweeterId;
  }
}
